mathics/core/expression.py

```diff
@@ -1897,7 +1897,11 @@ def _is_neutral_symbol(symbol_name, cache, evaluation):
 
     definitions = evaluation.definitions
 
-    definition = definitions.get_definition(symbol_name, only_if_exists=True)
+    try:
+        definition = definitions.get_definition(symbol_name, only_if_exists=True)
+    except KeyError:
+        definition = None
```

Comment:
maybe it would be better:
```
        cache[symbol_name] = True
        return True
```
to avoid the assignment to `definition:Definition` of a NoneType.

Reply:
Ok. Please make the change and commit. Thanks!

Reply:
Ok. Today I am going to go over again the code to see if there are more of this inconsistencies.